internal/services/iothub/iothub_endpoint_storage_container_resource.go

```diff
@@ -150,6 +151,14 @@ func resourceIothubEndpointStorageContainerSchema() map[string]*pluginsdk.Schema
 				string(devices.EncodingJSON),
 			}, true),
 		},
+
+		// NOTE: O+C : required since this property would always be set even if it isn't specified in the tf config, otherwise it would cause a diff and break existing users
+		"subscription_id": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ValidateFunc: validation.IsUUID,
```

Comment:
```suggestion
		"subscription_id": {
			Type:         pluginsdk.TypeString,
			Optional:     true,
			// NOTE: O+C : required since this property would always be set even if it isn't specified in the tf config, otherwise it would cause a diff and break existing users
			Computed:     true,
			ValidateFunc: validation.IsUUID,
```

Reply:
Updated